pkg/controller/template/template_controller.go

```diff
@@ -480,7 +524,13 @@ func (ctrl *Controller) syncControllerConfig(key string) error {
 		glog.V(2).Infof("%v", err)
 		return ctrl.syncFailingStatus(cfg, err)
 	}
-	mcs, err := getMachineConfigsForControllerConfig(ctrl.templatesDir, cfg, pullSecretRaw, fg)
+	storageConfig, err := ctrl.storageConfLister.Get(ctrlcommon.ClusterFeatureInstanceName)
+	if err != nil && !errors.IsNotFound(err) {
```

Comment:
I may have missed the conclusion of this discussion. If the object is not yet created during install, or we have some weird flake, we would bypass this and call the next function but with a (presumably) nil storageConfig. Would that cause any problems? Or is that fine/we won't run into those situations?

Reply:
we are defaulting to the CSI migration is the storage config is nil, i tried to remove this `IsNotFound` check but it broke the unit tests mightily, and i need more time to fix those. we decided that the effect was limited enough that we could work on this specific piece as part of a separate bug [OCPBUGS-11833](https://issues.redhat.com/browse/OCPBUGS-11833).